tools/generator/program.go

```diff
@@ -26,399 +37,280 @@ import (
+		fmt.Println("Formatted: ", formattedCount)
+		fmt.Println("Built: ", builtCount)
+		close(done)
 	}
-	clone := exec.Command("git", "clone", origin, cloneLoc)
-	clone.Stdout = os.Stdout
-	clone.Stderr = os.Stderr
-	return cloneLoc, clone.Run()
 }
 
-func checkoutAPISpecsVer(target, repoLocation string) error {
-	checkout := exec.Command("git", "checkout", target)
-	checkout.Stdout = os.Stdout
-	checkout.Stderr = os.Stderr
-	checkout.Dir = repoLocation
-	return checkout.Run()
-}
+func init() {
+	var useRecursive, useStatus bool
 
-// getDefaultOutputLocation returns the path to the local enlistment of the Azure SDK for Go.
-// If there is no local enlistment, it creates a temporary directory for the output.
-func getDefaultOutputLocation() string {
-	goPath := os.Getenv("GOPATH")
+	const automaticLogDirValue = "temp"
+	const logDirUsage = "The root directory where all logs can be found. If the value `" + automaticLogDirValue + "` is supplied, a randomly named directory will be generated in the $TEMP directory."
 
-	if goPath != "" {
-		sdkLocation := path.Join(goPath, "src", "github.com", "Azure", "azure-sdk-for-go")
-		if isGitDir(sdkLocation) {
-			return filepath.Clean(sdkLocation)
-		}
-	}
+	flag.BoolVar(&useRecursive, "r", false, "Recursively traverses the directories specified looking for literate files.")
+	flag.StringVar(&outputBase, "o", getDefaultOutputBase(), "The root directory to use for the output of generated files. i.e. The value to be treated as the go-sdk-folder when AutoRest is called.")
+	flag.BoolVar(&useStatus, "v", false, "Print status messages as generation takes place.")
+	flag.BoolVar(&dryRun, "p", false, "Preview which packages would be generated instead of actaully calling autorest.")
+	flag.StringVar(&logDirBase, "l", getDefaultOutputBase(), logDirUsage)
+	flag.Parse()
 
-	if tempDir, err := ioutil.TempDir("", "azure-sdk-for-go"); err == nil {
-		return filepath.Clean(tempDir)
+	statusWriter := ioutil.Discard
+	if useStatus {
+		statusWriter = os.Stdout
 	}
-	return "./"
-}
+	statusLog = log.New(statusWriter, "[STATUS] ", 0)
+	errLog = log.New(os.Stderr, "[ERROR] ", 0)
 
-func isGitDir(dir string) bool {
-	retval := false
-	if children, err := ioutil.ReadDir(dir); err == nil {
-		for _, child := range children {
-			if child.IsDir() && child.Name() == ".git" {
-				retval = true
-				break
-			}
+	if logDirBase == automaticLogDirValue {
+		var err error
+		logDirBase, err = ioutil.TempDir("", "az-go-sdk-logs")
+		logDirBase = normalizePath(logDirBase)
+		if err == nil {
+			statusLog.Print("Generation logs can be found at: ", logDirBase)
+		} else {
+			errLog.Print("Logging disabled. Could not create directory: ", logDirBase)
 		}
 	}
-	return retval
-}
 
-func generate(swag SwaggerFile, outputRootPath, specsRootPath string, output io.Writer) (outputDir string, err error) {
-	if output == nil {
-		output = ioutil.Discard
-	}
+	targetFiles = collection.AsEnumerable(flag.Args())
```

Comment:
>Args [](start = 44, length = 4)

Example?

Reply:
These are the arguments to the program. https://godoc.org/flag#Args

Is that what you mean?

Reply:
I mean an example of a program argument, what can I input as argument?

Reply:
I just added a new commit which replaces the `flag.Usage` global variable function, the new tells the caller that the expected arguments are literate files.